File: ejercicio3/orden_superior.scala
object Tribonacci{

  def fiboRec(x:Int):Int={
    if(x==0) return 0 
    else if(x==1) 1
    else if(x==2) 1
    else fibo(x-1) + fibo(x-2)
  }

  def fsuperior(fiR:(Int)=>Int):Int={
    return fiR(n)
  }

  def main(args:Array[String]):Unit={ 
    println("Dame un valor a: ")
    val n = scala.io.StdIn.readInt()  
    for(w <- 0 to n){
      print(fibo(w)+",")    
    } 
  } 

}

File: ejercicio3/orden_superior.py
n = 10
def fibo(x) :
    if (x == 0 ) :
        return 0
    if (x == 1) :
        return 0
    if (x == 2) :
        return 1
    return (fibo(x - 1) +fibo(x - 2) +fibo(x - 3))

def fsuperior(fun, x):
    return fun(x)

for i in range(1, n):
    print(f'{fsuperior(fibo, i)},', end='')
print(fsuperior(fibo, n))